### ISSUE TITLE
Adding text at end of link updates link text in Firefox but adds text after link in Chrome

### ISSUE THREAD

**Issue by heyitstowler** [liferay/alloy-editor#836]
## Prelude
I've enhanced my editor to style certain link classes to look like a call-to-action button, like so:

![cta](https://user-images.githubusercontent.com/16481282/39936666-2c7fd9de-551b-11e8-836f-78bb38b92a7a.png)

I've been doing most of my developing on Firefox before I realized that most of my content team uses chrome, so I switched to chrome. Unfortunately, there are a few behaviors that differ between the two browsers, and this one regarding link behavior is kind of a big one for me.

In Firefox, I could click the link and edit the text within the link as though I were editing the "button" text directly:
![cta1](https://user-images.githubusercontent.com/16481282/39937082-a296474c-551c-11e8-8ae5-2ba5165ec694.gif)
In chrome, however, the caret begins typing outside the link:
![cta2](https://user-images.githubusercontent.com/16481282/39937176-e85c6c0c-551c-11e8-98cf-9a0d44d66ef1.gif)

I've confirmed this is the normal behavior of links and not something specific to my setup, as the same behavior exists on the alloy editor demo:
Firefox: 
![mozilla](https://user-images.githubusercontent.com/16481282/39937353-6d92f878-551d-11e8-800c-30dcbb6fab23.gif)
Chrome: 
![chrome](https://user-images.githubusercontent.com/16481282/39937361-74c8dc3e-551d-11e8-9bf5-2428ca0849df.gif)

Is there a workaround to make the chrome behavior match the firefox behavior?  
Are there specific events/handlers you could point me at to investigate workarounds on my own?
I'm just not sure where to begin on this one, so I'm hoping someone who knows the core behavior better could chime in or at least point me in the right direction.

tl;dr when the caret is at the end of the link, firefox adds any additional text within the `<a></a>` element, whereas chrome adds it after the `<a></a>` element. Why do they behave differently, and how can I make chrome match Firefox's behavior?

## User story
Provide the Connextra styled explanation of the issue.
As a content editor
I'd like to type directly into the link
So that I can modify the text of my link without unlinking and relinking

## BDD Scenarios
Collect all the scenarios relevant to the applied changes and compose them in a Gherkin Syntax.

Scenario 1:
GIVEN the caret is at the end of a link tag
WHEN the user types additional content
THEN the additional content appears within the link tag
